math/cos_test.go

```diff
@@ -0,0 +1,44 @@
+package math
+
+import (
+	"math"
+	"testing"
+)
+
+const accuracy = 0.001
+
+func TestCos(t *testing.T) {
+	tests := []struct {
+		name string
+		n    float64
+		want float64
+	}{
+		{"cos(0)", 0, 1},
+		{"cos(90)", 90, -0.447},
+		{"cos(180)", 180, -0.598},
+		{"cos(1)", 1, 0.540},
+		{"cos(1)", math.Pi, -1},
+		{"cos(1)", math.Pi / 2, 0},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got := Cos(test.n)
+			if math.Abs(got-test.want) >= accuracy {
+				t.Errorf("Cos() = %v, want %v", got, test.want)
+				t.Errorf("MATH Cos() = %v", math.Cos(test.n))
+			}
+		})
+	}
+}
+
+func BenchmarkCos(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		Cos(180)
+	}
+}
+
+func BenchmarkMathCos(b *testing.B) {
```

Comment:
Just to point out that the reason why the Math Cos is slow is because the epsilon tolerance there is quite high. So I think you should add a comment here to make sure people are aware of this fact.

Reply:
Done